### PR TITLE
💥 FE dev environment setup

### PR DESCRIPTION
**Pull request by george-mzai**
image

## How to test it
-
   ```bash
 
   cd frontend
   npm install
   npm run dev
   ```

- Go to` localhost:3000` in your browser

## I already...
- Added [quasar](https://quasar.dev/) UI library

### CODE REVIEW

**Comment by chainlink** [comment]
Welcome @george-mzai! Curious why we're using Vue.js here over React

**Comment by george-mzai** [comment]
> Welcome @george-mzai! Curious why we're using Vue.js here over React

Thanks really excited to be here! 
The short answer would be the "open-source context" we are working into.

🛣️ Vue offers an easier learning curve. So a beginner-friendly setup potentially will attract more contributors.

🧽 Clean code. Vue’s single-file components allow for a clean separation of concerns (HTML, CSS, and JavaScript).
 
🛤️ Vue offers a more opinionated structure compared to React, but it still gives us flexibility. This helps maintain consistency across the project, reducing decision fatigue for contributors who might be unfamiliar with broader choices in the React ecosystem (like choosing between React's state management libraries or build tools).

🎁 Vue had an amazing year regarding updates(released 3.5) and new features with a big community behind it.

File: .DS_Store
Comment (veekaybee) [inline .DS_Store:1]: We'll want to add `.DS_store` to the .gitignore at the base of the repo so we don't commit this file. https://github.com/mozilla-ai/lumigator/blob/main/.gitignore

**Comment by chainlink** [comment]
> The short answer would be the "open-source context" we are working into.
Sorry I don't follow, React and other FE frameworks are also open source.

**Comment by george-mzai** [comment]
> > The short answer would be the "open-source context" we are working into.
> 
> Sorry I don't follow, React and other FE frameworks are also open source.

I am referring to Lumigator being open-source, not to the FE frameworks. Vue offers an opinionated yet flexible structure, which helps keep the project consistent without the need to make too many decisions.Also has more intuitive syntax which makes it easier for new contributors to jump in quickly, a valuable element for community-driven projects.

**Comment by agpituk** [comment]
@george-mzai I'm quite curious with the PR. It looks like code that was generated using the create project commands for Vue / Vite / Quasar.
There are things that we don't definitely need in the repo (those .sass variables that aren't called elsewhere, vue/vite logos, etc.)
Also we're using a HelloWorld component that looks out of a quickstart guide/tutorial 
`<button type="button" @click="count++">count is {{ count }}</button>`  with counters on buttons.

Not sure why we need any of this code in the repo in general instead of more focused lumigator code. Not sure how we far are with the designs and how clear is for you what we need to build, but if what you want is to give us some structure on how the code will be run in local development so we can help set up that process, I'd recommend going for a smaller approach that's tailored for lumigator and not a general quickstart Vite + Vue + Quasar project (instead of doing a helloworld just show a list of the datasets endpoint? a simple welcome page?)

**Comment by george-mzai** [comment]
> > The short answer would be the "open-source context" we are working into.
> 
> Sorry I don't follow, React and other FE frameworks are also open source.

I am referring to Lumigator being open-source, not to the FE frameworks. Vue offers an opinionated yet flexible structure, which helps keep the project consistent without the need to make too many decisions.Also has more intuitive syntax which makes it easier for new contributors to jump in quickly, a valuable element for community-driven projects.


> @george-mzai I'm quite curious with the PR. It looks like code that was generated using the create project commands for Vue / Vite / Quasar. There are things that we don't definitely need in the repo (those .sass variables that aren't called elsewhere, vue/vite logos, etc.) Also we're using a HelloWorld component that looks out of a quickstart guide/tutorial `<button type="button" @click="count++">count is {{ count }}</button>` with counters on buttons.
> 
> Not sure why we need any of this code in the repo in general instead of more focused lumigator code. Not sure how we far are with the designs and how clear is for you what we need to build, but if what you want is to give us some structure on how the code will be run in local development so we can help set up that process, I'd recommend going for a smaller approach that's tailored for lumigator and not a general quickstart Vite + Vue + Quasar project (instead of doing a helloworld just show a list of the datasets endpoint? a simple welcome page?)

Understood. I 'll move forward with the changes.

File: .DS_Store
Comment (agpituk) [reply .DS_Store:1]: @george-mzai can you add `.DS_store` to .[gitignore](https://github.com/mozilla-ai/lumigator/blob/main/.gitignore) as @veekaybee suggested?

File: lumigator/frontend/src/assets/vue.svg
Comment (agpituk) [inline lumigator/frontend/src/assets/vue.svg:1]: I don't think this needs to be in the repo

File: lumigator/frontend/.vite/deps_temp_2f4f4a8e/package.json
Comment (agpituk) [inline lumigator/frontend/.vite/deps_temp_2f4f4a8e/package.json:1]: I'm not an expert in vite at all but this looks like build files that shouldn't be in the repo as well?

File: lumigator/frontend/src/services/http/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios';
+
+const http = axios.create({
+  baseURL: 'http://localhost:8000/api/v1/', // API base URL
```
Comment (agpituk) [inline lumigator/frontend/src/services/http/index.js:4]: Can we make the baseURL be passed as an environment variable?

File: lumigator/frontend/src/services/http/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+export default http;
+
+// http://localhost:8000/api/v1/datasets/
```
Comment (agpituk) [inline lumigator/frontend/src/services/http/index.js:12]: remove unused commented lines

File: lumigator/frontend/src/components/Actions.vue [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<!-- eslint-disable vue/multi-word-component-names -->
```
Comment (chainlink) [inline lumigator/frontend/src/components/Actions.vue:1]: Is this file still in progress?

**Comment by george-mzai** [comment]
Will replace with leaner boilerplate code.

**Pull request closed without merging**